File: ForgeModSupportTab/src/main/java/com/n9mtq4/exmcl/tab/forgemods/ui/FModsTable.java
package com.n9mtq4.exmcl.tab.forgemods.ui;

import com.n9mtq4.exmcl.tab.forgemods.data.ModData;
import com.n9mtq4.exmcl.tab.forgemods.utils.FileDrop;

import javax.swing.JTable;
import javax.swing.ListSelectionModel;
import java.io.File;

/**
 * Created by will on 7/28/15 at 11:25 AM.
 */
public final class FModsTable extends JTable {
	
	private ModData modData;
	private ForgeTab forgeTab;
	private FModsTableModel model;
	
	public FModsTable(ModData modData, ForgeTab forgeTab) {
		
		super();
		
		this.modData = modData;
		this.forgeTab = forgeTab;
		this.model = new FModsTableModel(modData, forgeTab, this);
		
		setModel(model);
		
		model.fireSet();
		 
		getTableHeader().setReorderingAllowed(false);
		setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
		
		initFileDrop();
		
	}
	
	protected void fireModDataSync() {
		model.fireModDataSync();
	}
	
	private void initFileDrop() {
		
		new FileDrop(this, new FileDrop.Listener() {
			@Override
			public void filesDropped(File[] files) {
				for (File file : files) {
					modData.profiles.get(modData.selectedProfile).addMod(file);
				}
				refreshModel();
			}
		});
		
	}
	
	public final void refreshModel() {
		
		model.refresh();
		
	}
	
	public final void refreshTab() {
		
		forgeTab.refresh();
		
	}
	
	public final ModData getModData() {
		return modData;
	}
	
	public final ForgeTab getForgeTab() {
		return forgeTab;
	}
	
	@Override
	public final FModsTableModel getModel() {
		return model;
	}
	
}

File: ForgeModSupportTab/src/main/kotlin/com/n9mtq4/exmcl/tab/forgemods/ui/FModsTable.kt
package com.n9mtq4.exmcl.tab.forgemods.ui

import com.n9mtq4.exmcl.tab.forgemods.data.ModData
import com.n9mtq4.exmcl.tab.forgemods.utils.FileDrop
import java.io.File
import javax.swing.JTable
import javax.swing.ListSelectionModel

/**
 * Created by will on 11/4/15 at 5:55 PM.
 *
 * @author Will "n9Mtq4" Bresnahan
 */
public final class FModsTable(modData: ModData, forgeTab: ForgeTab) : JTable() {
	
	var modData: ModData
	val forgeTab: ForgeTab
	val forgeModel: FModsTableModel
	
	init {
		
		this.modData = modData
		this.forgeTab = forgeTab
		this.forgeModel = FModsTableModel(modData, forgeTab, this)
		
		model = forgeModel
		
		forgeModel.fireSet()
		
		tableHeader.reorderingAllowed = false
		setSelectionMode(ListSelectionModel.SINGLE_SELECTION)
		
		initFileDrop()
		
	}
	
	fun fireModDataSync() {
		forgeModel.fireModDataSync()
	}
	
	fun refreshModel() {
		forgeModel.refresh()
	}
	
	fun refreshTab() {
		forgeTab.refresh()
	}
	
	private fun initFileDrop() {
		
		FileDrop(this, FileDrop.Listener { files: Array<File> -> 
			files.forEach { modData.profiles[modData.selectedProfile].addMod(it) }
			refreshModel()
		})
		
	}
	
}
